data-transform.qmd

```diff
@@ -473,76 +467,73 @@ For example, the following code finds the most delayed flight to each destinatio
 
 ```{r}
 flights |> 
-  group_by(dest) |> 
-  slice_max(arr_delay, n = 1)
+  slice_max(arr_delay, n = 1, by = dest)
 ```
 
 This is similar to computing the max delay with `summarize()`, but you get the whole row instead of the single summary:
 
 ```{r}
 flights |> 
-  group_by(dest) |> 
-  summarize(max_delay = max(arr_delay, na.rm = TRUE))
+  summarize(
+    .by = dest,
+    max_delay = max(arr_delay, na.rm = TRUE),
+  )
 ```
 
-### Grouping by multiple variables
-
-You can create groups using more than one variable.
-For example, we could make a group for each day:
+You can also use `.by` with `filter()` to express the same idea in a slightly different way:
 
 ```{r}
-daily <- flights |>  
-  group_by(year, month, day)
-daily
+flights |> 
+  filter(
+    .by = dest,
+    arr_delay == max(arr_delay, na.rm = TRUE),
+  )
```

Comment:
How does one read this?

"Take the flights data frame and filter it, by each destination, for observations by where arrival delay is equal to the maximum arrival delay for that destination."

Since we didn't do `group_by()` first, I felt like "maximum arrival delay *for that destination*" is not immediately clear.

I also think `.by` should be the second argument but these are separate concerns.